src/lib.rs

```diff
@@ -18,3 +18,6 @@ pub mod perm_uses;
 
 /// Implementation for the Poseidon Sponge hash function
 pub mod sponge;
+
+/// Implementation of the Poseidon permutation based on the Hades strategy
+pub mod hades;
```

Comment:
Is pub necessary?

Reply:
That is indeed debatable! For this PR I wanted to simply port everything from hades without doing a lot of changes and since the crate was public, I though it makes sense to make the `hades` module in our poseidon library public as well. I opened an issue to only expose two permutation functions (one normal and the gadget) and the hades `WIDTH` #243

Reply:
I'll merge this PR and we continue the discussion in #243